src/go/k8s/internal/client/factory_test.go

```diff
@@ -0,0 +1,240 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/redpanda-data/helm-charts/pkg/helm"
+	"github.com/redpanda-data/helm-charts/pkg/kube"
+	"github.com/redpanda-data/helm-charts/pkg/testutil"
+	redpandav1alpha1 "github.com/redpanda-data/redpanda-operator/src/go/k8s/api/redpanda/v1alpha1"
+	redpandav1alpha2 "github.com/redpanda-data/redpanda-operator/src/go/k8s/api/redpanda/v1alpha2"
+	"github.com/redpanda-data/redpanda-operator/src/go/k8s/pkg/k3d"
+	"github.com/stretchr/testify/require"
+	"github.com/twmb/franz-go/pkg/kadm"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var chartVersion = ""
+
+func init() {
+	log.SetLogger(logr.Discard())
```

Comment:
Is this necessary when using context based plumbing?

Reply:
Yeah, unfortunately, I can get close, but even after passing in a context with `log.IntoContext(ctx, logr.Discard())` and attempting to silence my local controller runtime client used for testing by initializing it with `client.Options{WarningHandler: client.WarningHandlerOptions{SuppressWarnings: true}}`, I get the same stack trace noisy log from a call to the underlying resource fetching code in the `helm-charts` repo which, somewhere along the way [initializes a client](https://github.com/redpanda-data/helm-charts/blob/df79bbd7f93c8186836aa4a0f025601c0c7666d3/pkg/kube/ctl.go#L57) without the `SuppressWarnings` flag:

```
[controller-runtime] log.SetLogger(...) was never called; logs will not be displayed.
Detected at:
	>  goroutine 32 [running]:
	>  runtime/debug.Stack()
	>  	/Users/andrewstucki/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.4.darwin-arm64/src/runtime/debug/stack.go:24 +0x64
	>  sigs.k8s.io/controller-runtime/pkg/log.eventuallyFulfillRoot()
	>  	/Users/andrewstucki/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/log/log.go:60 +0xf4
	>  sigs.k8s.io/controller-runtime/pkg/log.(*delegatingLogSink).WithName(0x1400047a5c0, {0x1027cc3ea, 0x14})
	>  	/Users/andrewstucki/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/log/deleg.go:147 +0x34
	>  github.com/go-logr/logr.Logger.WithName({{0x1032c61f0, 0x1400047a5c0}, 0x0}, {0x1027cc3ea?, 0x1400083a780?})
	>  	/Users/andrewstucki/go/pkg/mod/github.com/go-logr/logr@v1.4.1/logr.go:345 +0x40
	>  sigs.k8s.io/controller-runtime/pkg/client.newClient(0x0?, {0x0, 0x0, {0x0, 0x0}, 0x0, {0x0, 0x0}, 0x0})
	>  	/Users/andrewstucki/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/client/client.go:129 +0xb4
	>  sigs.k8s.io/controller-runtime/pkg/client.New(0xf?, {0x0, 0x0, {0x0, 0x0}, 0x0, {0x0, 0x0}, 0x0})
	>  	/Users/andrewstucki/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/client/client.go:110 +0x54
	>  github.com/redpanda-data/helm-charts/pkg/kube.FromRESTConfig(0x140005d9448)
...
```

There's a reason I've always hated logging in Kubernetes and especially controller-runtime 😞